web/src/pages/api/public/traces/[traceId].ts

```diff
@@ -173,4 +178,81 @@ export default withMiddlewares({
       return { message: "Trace deleted successfully" };
     },
   }),
+
+  PATCH: createAuthedAPIRoute({
+    name: "Update Single Trace",
+    querySchema: PatchTraceV1Query,
+    bodySchema: PatchTraceV1Body,
+    responseSchema: PatchTraceV1Response,
+    fn: async ({ query, body, auth }) => {
+      const { traceId } = query;
+
+      // Get the trace to update
+      const trace = await getTraceById(traceId, auth.scope.projectId);
+
+      if (!trace) {
+        throw new LangfuseNotFoundError(
+          `Trace ${traceId} not found within authorized project`,
+        );
+      }
+
+      // Log audit entries for each updated field
+      const updates = [];
+
+      if (
+        body.bookmarked !== undefined &&
+        trace.bookmarked !== body.bookmarked
+      ) {
+        updates.push("bookmarked");
+        await auditLog({
+          resourceType: "trace",
+          resourceId: traceId,
+          action: "bookmark",
+          after: body.bookmarked,
+          projectId: auth.scope.projectId,
+          apiKeyId: auth.scope.apiKeyId,
+          orgId: auth.scope.orgId,
+        });
+        trace.bookmarked = body.bookmarked;
+      }
+
+      if (body.public !== undefined && trace.public !== body.public) {
+        updates.push("public");
+        await auditLog({
+          resourceType: "trace",
+          resourceId: traceId,
+          action: "publish",
+          after: body.public,
+          projectId: auth.scope.projectId,
+          apiKeyId: auth.scope.apiKeyId,
+          orgId: auth.scope.orgId,
+        });
+        trace.public = body.public;
+      }
+
+      if (
+        body.tags !== undefined &&
+        JSON.stringify(trace.tags) !== JSON.stringify(body.tags)
+      ) {
+        updates.push("tags");
+        await auditLog({
+          resourceType: "trace",
+          resourceId: traceId,
+          action: "updateTags",
+          after: body.tags,
+          projectId: auth.scope.projectId,
+          apiKeyId: auth.scope.apiKeyId,
+          orgId: auth.scope.orgId,
+        });
+        trace.tags = body.tags;
+      }
```

Comment:
Add additional /traces/<id>/tags/(tagId) routes for POST and DELETE to allow adding single tags and removing single tags.